lib/src/bindings/flatbuffers.dart

```diff
@@ -139,18 +144,43 @@ class OBXFlatbuffersManager<T> {
     return _entityBuilder(propVals);
   }
 
+  T /*?*/ unmarshalWithMissing(Pointer<Uint8> dataPtr, int length) {
+    if (dataPtr == null || dataPtr.address == 0 || length == 0) {
+      return null;
+    }
+    return unmarshal(dataPtr, length);
+  }
+
   // expects pointer to OBX_bytes_array and manually resolves its contents (see objectbox.h)
-  List<T> unmarshalArray(final Pointer<OBX_bytes_array> bytesArray,
-      {bool allowMissing = false}) {
+  List<T> unmarshalArray(final Pointer<OBX_bytes_array> bytesArray) {
     final result = <T>[];
     result.length = bytesArray.ref.count;
 
     for (var i = 0; i < bytesArray.ref.count; i++) {
       final bytesPtr = bytesArray.ref.bytes.elementAt(i);
-      if (allowMissing && (bytesPtr == nullptr || bytesPtr.ref.size == 0)) {
+      if (bytesPtr == null || bytesPtr.address == 0 || bytesPtr.ref.size == 0) {
+        throw ObjectBoxException(
+            dartMsg: "can't access data of empty OBX_bytes");
+      }
+      result[i] = unmarshal(bytesPtr.ref.data.cast<Uint8>(), bytesPtr.ref.size);
+    }
+
+    return result;
+  }
+
+  // expects pointer to OBX_bytes_array and manually resolves its contents (see objectbox.h)
+  List<T /*?*/ > unmarshalArrayWithMissing(
+      final Pointer<OBX_bytes_array> bytesArray) {
+    final result = <T /*?*/ >[];
+    result.length = bytesArray.ref.count;
+
+    for (var i = 0; i < bytesArray.ref.count; i++) {
+      final bytesPtr = bytesArray.ref.bytes.elementAt(i);
+      if (bytesPtr == nullptr || bytesPtr.ref.size == 0) {
```

Comment:
`bytesPtr == nullptr` is the same check actually. 

from `ffi.dart`
```
final Pointer<Never> nullptr = Pointer.fromAddress(0);

...

  /// Equality for Pointers only depends on their address.
  bool operator ==(Object other) {
    if (other is! Pointer) return false;
    Pointer otherPointer = other;
    return address == otherPointer.address;
  }
```

this passes in tests:

```
    var ptr = ffi.Pointer<ffi.Uint8>.fromAddress(0);
    expect(ptr, equals(ffi.nullptr));
```

Reply:
yeah and the function you're referring to (wasn't in the code preview above) is `unmarshalArrayWithMissing`, i.e. it "allows" missing values - used by getMany(). 

the matching code from the `main` was pretty much the same, except there was just one function with an "allowMissing" argument (can't be done anymore because the nullable type would bubble up to the public getMany()/getAll() while only the former should return a `List<T?>` while getAll() always has values, so returns `List<T>`

```dart
      if (allowMissing && (bytesPtr == nullptr || bytesPtr.ref.size == 0)) {
        result[i] = null;
      } else {
        result[i] = unmarshal(OBX_bytes_wrapper.safeDataAccess(bytesPtr));
      }
```

Reply:
Ah, I read `== nullptr` as `== null`. Then: why is there no null check on `bytesPtr` here, but there is in the `unmarshalArray` function above?

Reply:
You're right, they're misaligned, I've updated them. This whole file is going to need some more attention once ffi & ffigen migrate to null safety. Though that's true for the whole repo, but especially anything ffi/related.